### DISCUSSION TITLE
ModernBERT

### DISCUSSION THREAD

**Discussion by atelierJVA** [MaartenGr/BERTopic#2250] [Q&A] [answered]
Recently a new version of BERT, [ModernBERT](https://huggingface.co/blog/modernbert) has been released, and I have a couple of questions:

- Will this version improve Bertopic topic classification?
- The larger  context length (8K vs 512) will help with longer documents?
- How do we use it in Bertopic?

Thanks in advance.

**Comment by MaartenGr** [comment, marked as answer]
Thank you for sharing. It is a great foundational/base model! That said, we typically do not use base BERT-like for creating the embeddings; we fine-tune them instead. This likely means we need to wait until someone fine-tunes it specifically use contrastive learning to create embeddings that are more suited towards tasks like clustering. 

That said, if you want to use it as is, you can load it in a Transformers pipeline and use it in BERTopic: https://maartengr.github.io/BERTopic/getting_started/embeddings/embeddings.html#hugging-face-transformers

**Reply by atelierJVA** [reply]
Thanks for the answer, and for Bertopic. I will keep an eye on fine-tuned models based on ModernBERT and experiment with them when they become available

**Reply by MaartenGr** [reply]
Also note that typically the best place to look for embedding models is the [MTEB leaderboard](https://huggingface.co/spaces/mteb/leaderboard). It's not a perfect leaderboard, but if there are fine-tunes of ModernBERT through methods like contrastive learning, you will find them there.

**Comment by tomaarsen** [comment]
Some time has passed, and there's now a bunch of options!
Have a look: https://huggingface.co/models?library=sentence-transformers&other=modernbert

Some particularly promising ones are:
* https://huggingface.co/nomic-ai/modernbert-embed-base
* https://huggingface.co/lightonai/modernbert-embed-large
* https://huggingface.co/tasksource/ModernBERT-base-embed
* https://huggingface.co/Parallia/Fairly-Multilingual-ModernBERT-Embed-BE
* https://huggingface.co/jxm/cde-small-v2

The bottom 2 are extra custom. Just based on a guess, I reckon that the 2nd to last one will still work, and the last one won't.

--- 

- Tom Aarsen

**Reply by MaartenGr** [reply]
@tomaarsen Thanks for sharing! Now that I have you here, any recommendations on new default English/Multilingual models?

I'm considering the following for English models (good speed/acc balance):
* thenlper/gte-small
* BAAI/bge-small-en-v1.5

A Multilingual alternative is tricky as the default paraphrase model is already quite good for clustering purposes...
